elastic/kibana
# [Fleet] Register pages for custom integrations language clients in Fleet UI

Phase 1 of https://github.com/elastic/kibana/issues/136470

## Summary
Today, when clicking on one of the integrations from `language client` category, it just links to a page in the docs. 
This PR introduces a mechanism that allows the Custom Integrations plugin to register a page in the plugin start method that will then be displayed from Integrations app when clicking on the language client tile

![Screenshot 2022-09-16 at 15 05 42](https://user-images.githubusercontent.com/16084106/190649374-b3141b16-0a9c-4b66-b085-d78c209210d2.png)


#### On Custom Integrations side:
- Moved the [languageIntegrations](https://github.com/elastic/kibana/blob/2679970cd1f70a97a83f902a51dafaa60d008788/src/plugins/custom_integrations/server/language_clients/index.ts#L25) object under `common` to have it available both to sever and frontend and added two parameters:
   - `integrationsAppUrl`: the URL in integrations app that will display the custom component
   - `exportLanguageUiComponent`: a boolean that basically acts like a feature flag. If false, the integrations tile will keep linking to the docs, if true the will link to the new component. This can be removed in feature when this work is completed.
- Exported `languageClientsUiComponents` in public/plugin start method, that will allow to get the custom components from Fleet. It creates a component for each of the `languageIntegrations` entries that have `exportLanguageUiComponent` = true.
- Created an example react component `OverviewComponent`. 
**Note that this is just an example that displays the name of the client.** In phase 2 this component will be replaced with an actual overview.

#### On Fleet side:
- Added a new route `integrations/language_clients/pkgkey/overview`
- Added a component `CustomLanguagesOverview` that will render the react component exported from custom integrations. This component renders whatever is inside it, it just checks that the component exists otherwise redirects to `/integrations`

#### How to test locally
- Manually edit this [file](https://github.com/criamico/kibana/blob/a975353e949a11205e7378481440641a207214dd/src/plugins/custom_integrations/common/index.ts#L153-L271) : `javascript.exportLanguageUiComponent: false`
- Go to Integrations app and click on "Javascript Client" tile.
- It should navigate to `integrations/language_clients/javascript/overview` and you should see "javascript client overview" on screen

![Screenshot 2022-09-16 at 15 29 45](https://user-images.githubusercontent.com/16084106/190650391-46225745-3f26-4058-b506-cf9ec5babfee.png)

- Go back to integrations app and click on any other language client tile
- It should still link to the docs

### Checklist
